src/Features/Lsif/Generator/Generator.cs

```diff
@@ -86,7 +88,21 @@ public void GenerateForCompilation(Compilation compilation, string projectPath,
             var syntaxFactsService = languageServices.GetRequiredService<ISyntaxFactsService>();
             var semanticFactsService = languageServices.GetRequiredService<ISemanticFactsService>();
 
-            var documentVertex = new Graph.LsifDocument(new Uri(syntaxTree.FilePath), GetLanguageKind(semanticModel.Language), idFactory);
+            string? contentBase64Encoded = null;
+
+            // TODO: move to checking the enum member mentioned in https://github.com/dotnet/roslyn/issues/49326 when that
+            // is implemented. In the mean time, we'll use a heuristic of the path being a relative path as a way to indicate
+            // this is a source generated file.
+            if (!PathUtilities.IsAbsolute(syntaxTree.FilePath))
+            {
+                var text = semanticModel.SyntaxTree.GetText();
+
+                // We always use UTF-8 encoding when writing out file contents, as that's expected by LSIF implementations.
```

Comment:
This is from talking with @BertanAygun, although I'm asking for further spec clarification in https://github.com/microsoft/language-server-protocol/issues/1139.

Reply:
Current answer at least is this is correct, so if that changes we'll come back and fix it.